eng/common/scripts/Package-Properties.ps1

```diff
@@ -237,6 +247,38 @@ function Get-PrPkgProperties([string]$InputDiffJson) {
                     if ($includedForValidation) {
                         $pkg.IncludedForValidation = $true
                     }
+                    break
+                }
+            }
+
+            # handle service-level changes to the ci.yml files
+            # we are using the ci.yml file being added automatically to each artifactdetails as the input
+            # for this task. This is because we can resolve a service directory from the ci.yml, and if
+            # there is a single ci.yml in that directory, we can assume that any file change in that directory
+            # will apply to all packages that exist in that directory.
+            if ($pathComponents.Length -eq 3 -and $pathComponents[0] -eq "sdk") {
```

Comment:
This assumption breaks in a few edge cases, namely go (e.g. https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/aztables/ci.yml)

Reply:
Yeah. This `if` check will mean we never evaluate this piece of code. I'm going to move this logic within the $triggeringPaths iteration. We're kinda uselessly iterating over the triggering paths again here.

Reply:
Ok. This version is checking out.

```
  "ChangedFiles": [
    "sdk/core/tests.yml"
  ],
```

is successfully triggering the `core` packages that belong to the single `core/ci.yml`.